flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/lineage/LineageFacetProvider.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.flink.connector.kafka.lineage;
+
+import org.apache.flink.streaming.api.lineage.LineageDatasetFacet;
+
+import java.util.List;
+
+/**
+ * Contains method which can be used for lineage schema facet extraction. Useful for classes like
+ * topic selectors or serialization schemas to extract dataset information from.
+ */
+public interface LineageFacetProvider {
```

Comment:
I feel this interface can be moved to flink core repo.

Reply:
I think so too. To me, it makes more sense to add it first into flink-core and remove here later when upgrading flink-core for flink-connector-kafka. This shouldn't block lineage interface implementation.